openscm/scmdataframebase.py

```diff
@@ -827,6 +827,107 @@ def r(*args, **kwargs):
         x._resample_cls = get_resampler(self)
         return x.resample(time=rule, **kwargs)
 
+    def quantile_over(self, cols, quantile):
+        """
+        Take the quantile over the input columns.
+
+        Here the quantile means the value of the data at a given point in the
+        cumulative distribution of values at each point in the timeseries, for each
+        timeseries once the groupby is applied. As a result, using ``quantile=0.5`` is
+        is the same as taking the median and not the same as taking the mean/average.
+
+        Parameters
+        ----------
+        cols : str, list of str
+            Columns to take the quantile over. The timeseries will be grouped by all
+            other columns in ``self.meta``
+
+        quantile : float
+            The quantile to return. Must be between 0 and 1
+
+        Returns
+        -------
+        pd.DataFrame
+            The quantiles of the timeseries, grouped by all columns in ``self.meta``
+            other than ``cols``
+        """
+        cols = [cols] if isinstance(cols, str) else cols
+        ts = self.timeseries()
+        group_cols = list(set(ts.index.names) - set(cols))
+
+        return ts.groupby(group_cols).quantile(quantile)
```

Comment:
FYI pd.core.groupby.DataFrameGroupBy.quantile also accepts a list of floats to define quantiles